rust/src/parser.rs

```diff
@@ -26,6 +26,37 @@ use crate::applayer;
 use std::os::raw::{c_void,c_char,c_int};
 use crate::applayer::{AppLayerGetTxIterTuple};
 
+#[repr(C)]
+pub struct AppLayerResult {
```

Comment:
@jasonish thoughts about moving this to `applayer.rs` instead?

Reply:
Yes. In fact I think parser and app-layer should probably be merged, or something.. They're all app-layer.

Reply:
Ok will look at merging them.